features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/voicemessages/timeline/VoiceMessagePresenter.kt

```diff
@@ -73,24 +90,37 @@ class VoiceMessagePresenter @AssistedInject constructor(
     @Composable
     override fun present(): VoiceMessageState {
 
-        val scope = rememberCoroutineScope()
+        DisposableEffect(Unit) {
+            Timber.d("ABAB ${this@VoiceMessagePresenter}: Composed")
+            onDispose {
+                Timber.d("ABAB ${this@VoiceMessagePresenter}: Decomposed")
+            }
+        }
 
         val playerState by player.state.collectAsState(VoiceMessagePlayer.State(isPlaying = false, isMyMedia = false, currentPosition = 0L))
```

Comment:
I have the feeling we should let the `Player` hoisting the additional values if necessary instead of adding all this complexity?

Reply:
Why? It should be injected and kept in memory in the `RoomScope`, it shouldn't be linked to the Composition scope

Reply:
`MediaPlayer` is kept in the RoomScope. That is where the ExoPlayer instance lives.

`VoiceMessagePlayer` is just a tiny wrapper around `MediaPlayer` that creates a "view" of the mediaplayer for every single voice message.

Reply:
Yes I had that, but is it not easier to make it RoomScoped so we can hoist state instead of adding another "data source"?

Reply:
> Yes I had that, but is it not easier to make it RoomScoped so we can hoist state instead of adding another "data source"?

We could do that, but we would have the same problem you mentioned in https://github.com/vector-im/element-x-android/pull/1743/files#r1383412061
Plus, what if some other presenter wants to persist some state that is not related to the media player?